lib/http/response/body.rb

```diff
@@ -53,7 +53,10 @@ def to_s
           @streaming  = false
           @contents   = String.new("").force_encoding(encoding)
 
-          while (chunk = @stream.readpartial)
+          length = @length || Float::INFINITY
```

Comment:
@ixti agree, but how should I "import" the length? The initializer signature changed, and I lost my assumptions. Just in case this PR takes again longer than planned: what do you suggest as the long term way to pass the content length from the response to the body?

When I implemented it, I changed the signature to use kwargs and added the encoding and content length as optional parameters. Can I go back to that strategy? as @tarcieri said, this is anyway planned to a major bump, so in theory I could.